catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -441,6 +366,17 @@ private static void updateScriptsToTreeStructure(Project project) {
 		}
 	}
 
+	private static boolean checkValidLanguageVersion(Float languageVersion) {
+		if (languageVersion <= CURRENT_CATROBAT_LANGUAGE_VERSION) {
+			if (languageVersion == 0.8f) {
+				return true;
+			}
```

Comment:
pls research if it is possible to include the if statement in the regex

Reply:
the regex seems ok, since there proably wasnt a published 0.81 i guess, but all in all wouldnt this read a little better as:

`return languageVersion == 0.8f || 
            (languageVersion <= CURRENT_CATROBAT_LANGUAGE_VERSION &&
                       languageVersion >= 0.9f)`

Reply:
yes, that looks better. I'd just turn the second part around (just to be extra nitpicky) so that it says: either 0.8f or greater than 0.9f and below current language version.

`return languageVersion == 0.8f || (languageVersion >= 0.9f && languageVersion <= CURRENT_CATROBAT_LANGUAGE_VERSION)`

Reply:
I removed the regex entirely, my thought was that, it works for now but, once the language version reaches 1.0 the regex in it's current form would allow invalid versions. So the regex would have to be dynamic, something like `mystring.matches("[" + x + " ][.][" + y + "]+..."` where x would be the first char in the 0.9 string and y the third one. Which would make it harder to read in the first place and we would have to remember all the highest versions per subversion so 0.8, 0.99992 if we would switch from 0.99992 to 1.0 etc. So I added a map which has all known versions, to check we only have to access 1 entry and see if it is null. (assuming the java map is as efficient as c++ maps)